<?xml version="1.0"?>
<launch>		

	<!-- Get jostick input--> 
	<node respawn="true" pkg="joy" type="joy_node" name="turtle_joy" >
		<param name="dev" type="string" value="/dev/input/js0" />		
	</node>
	
	<!-- Run with jostick-->    
	<node pkg="self_driving_rc_car" type="joystick_controller" name="teleop_rc_car" respawn="true"></node>
    <node pkg="self_driving_rc_car" type="dbw.py" name="dbw_listener" respawn="true"></node>

	<!-- Run jetson camera using jetson-inference and jetbot-ros -->
	<node pkg="jetbot_ros" type="jetbot_camera" name="jetbot_camera"></node>

	<!-- Convert images from raw to compressed -->
	<!-- <node pkg="image_proc" type="image_proc" name="image_proc" >
		<remap from="image_raw" to="jetbot_camera/raw"/>
	</node> -->

	<!-- TF publisher -->
	<include file="$(find rc_car_description)/launch/tf_static.launch"/>
	<!-- BMI160 IMU Publisher-->
	<node pkg="self_driving_rc_car" type="imu_bmi160_filtered_publisher" name="imu_publisher"></node>
	<!-- RPLIDAR Launch -->
	<include file="$(find rplidar_ros)/launch/rplidar.launch"/>
	<!-- Wheel Encoder -->
    <node pkg="self_driving_rc_car" type="wheel_encoder.py" name="wheel_encoder" respawn="true"></node>
	<!-- EKF -->
	<!-- <node pkg="self_driving_rc_car" type="ext_kf" name="ext_kf"></node> -->




</launch>
